CITATION.cff

```diff
@@ -0,0 +1,9 @@
+cff-version: 1.2.0
+message: If you use this guide, please cite it using these metadata.
+title: Rustc Dev Guide
+abstract: A guide to developing rustc
+authors:
+  - name: "Rustc developers"
```

Comment:
I would not do `Rustc`, but rather `rustc` or `Rust compiler`

Reply:
I forget, do we still use "The Rust Project Developers" as a generic author name? Or was that phased out?

Reply:
> I forget, do we still use "The Rust Project Developers" as a generic author name? Or was that phased out?

I haven't seen any discussion about deprecating it and I'd assume we still use. I also think it's a better name here, let's use it.

Reply:
Updated to "The Rust Project Developers".